### ISSUE TITLE
Please please help with a problem with this source (https://chrysanthemumgarden.com/)

### ISSUE THREAD

**Issue by norabelle101** [dteviot/WebToEpub#657]
Hello!

I'm so sorry for the trouble and I know this isn't a problem or bug with the software itself, but I was wondering if I could be advised on how to bypass a certain big problem on this source.

When I try to convert any chapters on this source (https://chrysanthemumgarden.com/) to EPUB, there's a glaring problem. **For every single "Chapter", three or four paragraphs are randomly turned into gibberish because of the site's copy protection.** 

**They're normal words on the site itself, but any means of "copying"/"parsing" them out of the site will result in being turned to gibberish (including "save as PDF").** Since I don't have access to internet all the time, it's more convenient reading as a PDF, so I've been desperately trying to find some way to overturn this. Please please, may I ask if this can be bypassed so I can create my EPUB with no issues?

Here's an example for reference too! Thank you so very much for any help! 

![image](https://user-images.githubusercontent.com/55352222/143796538-dab3f27d-2eff-4fc0-bc61-62e28176b22b.png)

**Comment by dteviot** [comment]
@vanessabluellie 
This is a repeat of https://github.com/dteviot/WebToEpub/issues/354

Short answer:  
1. Go to https://drive.google.com/drive/folders/1B_X2WcsaI_eg9yA-5bHJb8VeTZGKExl8?usp=sharing and download EpubEditor.zip
2. Unpack the zip and open the file main.html in chrome.
3. Drag and drop Chrysanthemum Garden epub onto the drop zone.
4. A set of buttons will appear, click the second from bottom labelled "Clean ChrysanthemumGarden"
5. Wait a little bit.  You should get a file dialog asking you for name to save the fixed epub.

Slightly longer answer.
Basically, in order to prevent the chapters on site being copied, the site scrambles the text and uses a special font to unscramble the text.  Because this is done as a copyright mechanism, if WebToEpub was to unscramble the text it would get WebToEpub removed from the chrome store.
However, I can provide a tool to unscramble the file that's not on the Chrome store.

 If you have any problems, please email me at dteviot@gmail.com

**Comment by vanessabluellie** [comment]
Hello! Thank you! But there's a little error still. Once I dragged the EPUB to that drop area and then click on "Clean ChrysanthemumGarden", it immediately downloads the same EPUB file with those scrambled words and doesn't seem to run the Epub editor tool at all... It also did the same on Firefox browser. Hope that I can be advised on how to fix this error! Thank you! And so sorry for the inconvenience.

**Comment by dteviot** [comment]
@vanessabluellie 
Can I have the epub file you tried?

**Comment by dteviot** [comment]
@vanessabluellie 
I'm not sure what you mean by 
> it immediately downloads the same EPUB file 

EpubEditor does not download anything.  It just processes the epub you dropped onto the form.
Once it has finished processing the epub, you should get a "save as" dialog to write the new epub.
Note, due to how chrome/firefox work, I needed to use the download API to save the file.

So, at the "Save as" dialog, give a different name (so you can tell the difference) and then try to open the new epub file.

**Comment by vanessabluellie** [comment]
> @vanessabluellie Can I have the epub file you tried?

Sure! I tried two different webnovel converters and tried using the epub editor with both files, but the result was the same. I tried various other novels and it still didn't run the editor. I'll attach them both! I couldn't upload them here, but I'll put them in Google Drive and here's the link! Thank you again so very much!

https://drive.google.com/drive/folders/1F1Dm6KQplYw89TZBrkOKB5PKpQBZ51_M?usp=sharing

**Comment by vanessabluellie** [comment]
> @vanessabluellie I'm not sure what you mean by
> 
> > it immediately downloads the same EPUB file
> 
> EpubEditor does not download anything. It just processes the epub you dropped onto the form. Once it has finished processing the epub, you should get a "save as" dialog to write the new epub. Note, due to how chrome/firefox work, I needed to use the download API to save the file.
> 
> So, at the "Save as" dialog, give a different name (so you can tell the difference) and then try to open the new epub file.

Oh, that "Save as" dialog never pops up at all. I meant that after I dropped the epub into the epub editor and then the drop down menu appears, I clicked on the option "Clean ChrysanthemumGarden", but the editor doesn't appear to run to clean the epub file at all and just immediately downloads the same file with the errors. I apologise for my inadequate explanation.

**Comment by dteviot** [comment]
@vanessabluellie 
Seems to work fine for me.  I've dropped the cleaned files to https://drive.google.com/drive/folders/1dmXUiSx0UmoPMJDF8hF3xFNu9wkrYCV-

When you say
>editor doesn't appear to run to clean the epub file at all and just immediately downloads the same file with the errors.

There isn't anything to see.  The dialog appears when cleaning is done.  Which only took a couple of seconds on my PC.

Some screenshots that might make it clearer.
1. Open Epub Editor
![image](https://user-images.githubusercontent.com/5157081/143999188-bde5b1d3-4e6f-41e8-9171-2568c27faaad.png)

2. Drop file 
![image](https://user-images.githubusercontent.com/5157081/143999320-201851af-9155-4e6c-98d1-d097b4896d0e.png)


3. Then click "Clean ChrysanthemumGarden" button.  A couple of seconds later, will be finished, and dialog for filename for cleaned epub will appear.
![image](https://user-images.githubusercontent.com/5157081/143999696-0947182b-7d64-479e-9d8b-286d14b1d6c1.png)
At this point, make sure filename ISN'T same as original, and click "Save" button.
Fixed epub will be written to new filename.

**Comment by vanessabluellie** [comment]
AH, the "Save As" dialog still doesn't pop up, but for some reason, it returned me the edited EPUB. It works perfectly now! Thank you!

**Comment by shonetypto** [comment]
> 3\. At this point, make sure filename ISN'T same as original, and click "Save" button.

Hello, 

I'm testing out a few things and i couldn't get epub editor to work until i saw this comment. Perhaps you could include this in the readme file for epub? It isn't explicitly clear. Without creating a "to be cleaned" copy, the Editor doesn't respond.

**Comment by dteviot** [comment]
@shonetypto 

1. Can you be more specific?  Which readme file are you referring to?
2. Even better, can you make the change and submit a pull request?  Note, pull request should use the https://github.com/dteviot/WebToEpub/tree/ExperimentalTabMode branch.

**Comment by lunevale** [comment]
> Once it has finished processing the epub, you should get a "save as" dialog to write the new epub.

Hi @dteviot , I know this is an old issue but the 'save as' dialog doesn't pop up for me too. As a result, the epub file that I dropped into Epub Editor still contains the scrambled words. I've tried it in Firefox and Chrome with different epub files but the same thing happens. Do you have any idea how to work around this? Many thanks in advance!

**Comment by dteviot** [comment]
@lunevale 
I don't know what's going wrong.
That said, if you are willing, I could make a version with some logging, get you to run that, and then look at the logs to see if they show anything.

**Comment by vanessabluellie** [comment]
@lunevale 

I think I had this problem before too. Because of my Chrome settings and the "Save-As" dialogue was disabled, when I saved files from EpubEditor, I had saved the NEW epub file in the SAME location as the scrambled text epub to which it just downloaded that old epub (with the same filename) with the suffix of (1) but the content of the scrambled text would stay the same and wouldn't be replaced correctly with the edited fixed epub.

Try maybe first dragging the scrambled text epub (NOT placed in your downloads folder) to EpubEditor and then process it through "Clean ChrysanthemumGarden" again. When the file automatically downloads into your downloads folder, there should be no file of the same filename to hinder it. Open the new epub and it should be the EDITED epub file with NO scrambled text now.

Or, just go to your Chrome browser settings, and under the Downloads tab, toggle the button for "Ask where to save files" to ON and when you process the file in EpubEditor, name the new epub to something completely different from the scrambled text epub. This should also fix the problem.

![Screenshot_20230209-075049_Chrome](https://user-images.githubusercontent.com/55352222/217653878-8e0c8266-908e-4622-88d5-0424d0471901.jpg)

**Comment by dteviot** [comment]
Update documentation

**Comment by lunevale** [comment]
@vanessabluellie 

> Try maybe first dragging the scrambled text epub (NOT placed in your downloads folder) to EpubEditor and then process it through "Clean ChrysanthemumGarden" again. When the file automatically downloads into your downloads folder, there should be no file of the same filename to hinder it. Open the new epub and it should be the EDITED epub file with NO scrambled text now.

At the time of writing my comment above, I did this and the returned file still contained scrambled words. BUT I'm glad to report that this method works now!

> Or, just go to your Chrome browser settings, and under the Downloads tab, toggle the button for "Ask where to save files" to ON and when you process the file in EpubEditor, name the new epub to something completely different from the scrambled text epub. This should also fix the problem.

Yes, this works too! Thank you so much!
